stackslib/src/chainstate/stacks/db/blocks.rs

```diff
@@ -4689,6 +4755,68 @@ impl StacksChainState {
         Ok(parent_miner)
     }
 
+    fn set_signer_key_authorization(
+        clarity_tx: &mut ClarityTx,
+        sender: &StacksAddress,
+        reward_addr: &TupleData,
+        num_cycles: u128,
+        pox_reward_cycle: u64,
+        signer_key_value: &Vec<u8>,
+        max_amount: Value,
+        auth_id: Value,
+        mainnet: bool,
+        active_pox_contract: &str,
+    ) -> Result<(), String> {
+        let signer_auth_args = vec![
+            Value::Tuple(reward_addr.clone()),
+            Value::UInt(num_cycles),
+            Value::UInt(u128::from(pox_reward_cycle)),
+            Value::string_ascii_from_bytes(Pox4SignatureTopic::StackStx.get_name_str().into())
+                .unwrap(),
+            Value::buff_from(signer_key_value.clone()).unwrap(),
+            Value::Bool(true),
+            max_amount,
+            auth_id,
+        ];
+
+        match clarity_tx.connection().as_transaction(|tx| {
+            tx.run_contract_call(
+                &sender.clone().into(),
+                None,
+                &boot_code_id(active_pox_contract, mainnet),
+                "set-signer-key-authorization",
+                &signer_auth_args,
+                |_, _| false,
+            )
+        }) {
+            Ok((value, _, _)) => {
+                if let Value::Response(ref resp) = value {
+                    if !resp.committed {
```

Comment:
This shouldn't be an error.  If the transaction contract-call fails, we still want to capture it in a transaction receipt (just as we do today).  You can see an example of this in `process_stacking_ops` in how we handle the call to `stack-stx`.

Reply:
Resolving as this code was removed